pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -524,3 +561,202 @@ func validateAzure(h *validatorHandler, m *Machine) (bool, utilerrors.Aggregate)
 	}
 	return true, nil
 }
+
+type gcpDefaulter struct {
```

Comment:
the other choice would be to embed this into `defaulterHandler`. Whatever you feel is more simple/readable works for me

Reply:
I wanted it this way as I think it will fit better with the MachineSet PR this way, I've removed the deafulterHandler from the function signatures to make it reusable